pkg/network/tracer/tracer_linux_test.go

```diff
@@ -69,13 +69,15 @@ func doDNSQuery(t *testing.T, domain string, serverIP string) (*net.UDPAddr, *ne
 	queryMsg.SetQuestion(dns.Fqdn(domain), dns.TypeA)
 	queryMsg.RecursionDesired = true
 	dnsClient := new(dns.Client)
-	dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
-	require.NoError(t, err)
-	defer dnsConn.Close()
-	dnsClientAddr := dnsConn.LocalAddr().(*net.UDPAddr)
-	_, _, err = dnsClient.ExchangeWithConn(queryMsg, dnsConn)
-	require.NoError(t, err)
-
+	var dnsClientAddr *net.UDPAddr
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
+		assert.Nil(c, err, "expected no DNS dial error")
```

Comment:
You should return early here if you get an error; `dnsConn` is likely `nil` and would panic on the next line. Also, use `assert.NoError`. You an use the return value of the assert to return early.

Reply:
based on the docs for `EventuallyWithT`, this should already "return early" by starting over the eventually on the next try. mentioned here: https://github.com/DataDog/datadog-agent/pull/20444#discussion_r1377729785